src/lua/init.c

```diff
@@ -981,10 +981,13 @@ lua_require_lib(lua_State *L, const char *libname)
  * Push argument and call a function on the top of Lua stack
  */
 static int
-lua_main(lua_State *L, int argc, char **argv)
+lua_main(lua_State *L, bool is_debugging, int argc, char **argv)
```

Comment:
You have `is_debugging` and `debugging` variables. Do you really need both?

Reply:
Yup. `is_debugging` is needed in `lua_main` to make sure that we are pushing script name to the Lua stack before calling `luadebug.lua::start`, but `debugging` is an extracted `opt_mask` flag in `run_script_f` function, similar to `bytecode` and `interactive`. 

Yes, they serve similar purpose but in different functions, and not, I could not rename `debugging` to `is_debugging` inside of `run_script_f` because it would be inconsistent with other flags naming (`bytecode` and `interactive`).

[Originally they were named exactly the same, but you've asked me to rename it in `lua_main` now you the problem]